WeakAuras/GenericTrigger.lua

```diff
@@ -2931,35 +2933,35 @@ do
           mh_dur = mh_rem and mh_rem / 1000;
           mh_name = mh_exp and getTenchName(mh) or "None";
           mh_icon = GetInventoryItemTexture("player", mh)
-          WeakAuras.ScanEvents("MAINHAND_TENCH_UPDATE");
         end
         if(math.abs((oh_exp or 0) - (oh_exp_new or 0)) > 1) then
           oh_exp = oh_exp_new;
           oh_dur = oh_rem and oh_rem / 1000;
           oh_name = oh_exp and getTenchName(oh) or "None";
           oh_icon = GetInventoryItemTexture("player", oh)
-          WeakAuras.ScanEvents("OFFHAND_TENCH_UPDATE");
         end
+        WeakAuras.ScanEvents("TENCH_UPDATE");
         WeakAuras.StopProfileSystem("generictrigger");
       end
 
       tenchFrame:SetScript("OnEvent", function(self, event, arg1)
         WeakAuras.StartProfileSystem("generictrigger");
-        if(arg1 == "player") then
-          timer:ScheduleTimer(tenchUpdate, 0.1);
+        if (event == "UNIT_INVENTORY_CHANGED" and arg1 == "player") then
+          tenchUpdate()
```

Comment:
i believe there was a reason to add a 0.1s delay before calling tenchUpdate, why do you remove it ?